### PR TITLE
doc: document transition to default Thread 1.3

### PR DESCRIPTION
**Pull request by edmont**
Make Thread 1.3.0 support clear.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

**Review by canisLupus1313 — APPROVED** [review]

File: doc/nrf/ug_thread_configuring.rst [hunk 1]
```diff
@@ -116,11 +116,12 @@ For more configuration options, read the module documentation.
 Thread Specification options
 ============================
 
-The OpenThread stack can be configured to operate in compliance with either the Thread 1.1 Specification  or the :ref:`Thread 1.2 Specification <thread_ug_supported_features_v12>`.
+The OpenThread stack can be configured to operate in compliance with either the Thread 1.1 Specification, the :ref:`Thread 1.2 Specification <thread_ug_supported_features_v12>` or the :ref:`Thread 1.3 Specification <thread_ug_supported_features_v13>`.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_configuring.rst:119]:
```suggestion
The OpenThread stack can be configured to operate in compliance with either the Thread 1.1 Specification, the :ref:`Thread 1.2 Specification <thread_ug_supported_features_v12>`, or the :ref:`Thread 1.3 Specification <thread_ug_supported_features_v13>`.
```

File: doc/nrf/ug_thread_supported_features.rst [hunk 2]
```diff
@@ -7,7 +7,7 @@ Supported Thread features
    :local:
    :depth: 2
 
-The OpenThread implementation of the Thread protocol supports all features defined in the Thread 1.3.0 Specification:
+The OpenThread implementation of the Thread protocol supports all features defined in the Thread 1.3.0 Specification which are required for the Thread 1.3 Certification program:
```
Comment (greg-fer) [inline doc/nrf/ug_thread_supported_features.rst:10]:
```suggestion
The OpenThread implementation of the Thread protocol supports all features defined in the Thread 1.3.0 Specification that are required for the Thread 1.3 Certification program:
```

File: doc/nrf/ug_thread_supported_features.rst [hunk 3]
```diff
@@ -30,6 +30,10 @@ The OpenThread implementation of the Thread protocol supports all features defin
 
   * Service Registration Protocol (SRP) client
 
+In |NCS|, you can choose which version of the Thread protocol to use in your application.
+By default, |NCS| supports Thread 1.3, which includes Thread 1.2 support.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_supported_features.rst:34]:
```suggestion
By default, |NCS| supports Thread 1.3, which includes support for Thread 1.2.
```

File: doc/nrf/ug_thread_supported_features.rst [hunk 4]
```diff
@@ -115,3 +116,21 @@ Limitations for Thread 1.2 support
...
+Thread 1.3 features
+*******************
+
+For more information about this Thread version, see the official `Thread 1.3.0 Features White Paper`_.
+
+DNS-based Service Discovery
+===========================
+
+Thread 1.3 Specification introduces DNS-SD Service Registration Protocol as a way for devices to advertise the fact that they provide services, while avoiding the use of multicast in the discovery.
+|NCS| provides the required SRP client functionality
```
Comment (greg-fer) [inline doc/nrf/ug_thread_supported_features.rst:131]:
```suggestion
|NCS| provides the required SRP client functionality.
```

File: doc/nrf/ug_thread_supported_features.rst [hunk 5]
```diff
@@ -115,3 +116,21 @@ Limitations for Thread 1.2 support
...
+
+Thread 1.3 features
+*******************
+
+For more information about this Thread version, see the official `Thread 1.3.0 Features White Paper`_.
+
+DNS-based Service Discovery
+===========================
+
+Thread 1.3 Specification introduces DNS-SD Service Registration Protocol as a way for devices to advertise the fact that they provide services, while avoiding the use of multicast in the discovery.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_supported_features.rst:130]:
```suggestion
Thread 1.3 Specification introduces DNS-SD Service Registration Protocol, which lets devices advertise the fact that they provide services while avoiding the use of multicast in the discovery.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
Couple of comments.

**Review by LuDuda — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**